compiler/semantic/analyzer.go

```diff
@@ -40,13 +40,14 @@ func Analyze(ctx context.Context, p *parser.AST, env *exec.Environment, extInput
 }
 
 type analyzer struct {
-	ctx     context.Context
-	files   *srcfiles.List
-	opStack []*ast.OpDecl
-	outputs map[*dag.Output]ast.Node
-	env     *exec.Environment
-	scope   *Scope
-	sctx    *super.Context
+	ctx      context.Context
+	files    *srcfiles.List
+	opStack  []*ast.OpDecl
+	cteStack []*ast.SQLCTE
```

Comment:
Nit: I think what's going on here would be substantially clearer if this were a cte slice.

```suggestion
	cteStack []*cte
```